Assets/Scripts/ConnectionManagement/ConnectionState/OfflineState.cs

```diff
@@ -27,7 +27,10 @@ class OfflineState : ConnectionState
         public override void Enter()
         {
             m_LobbyServiceFacade.EndTracking();
-            m_ConnectionManager.NetworkManager.Shutdown();
+            if (m_ConnectionManager.NetworkManager.IsListening)
```

Comment:
This would be the fix for the issue with m_ShuttingDown being set to true again.
We might be able to add one more simple fix into the v1.5.2 branch if it isn't too late and then we can revert these updates.